packages/flutter_test/lib/src/finders.dart

```diff
@@ -460,57 +469,544 @@ class CommonFinders {
   }
 }
 
-/// Searches a widget tree and returns nodes that match a particular
-/// pattern.
-abstract class Finder {
-  /// Initializes a Finder. Used by subclasses to initialize the [skipOffstage]
-  /// property.
-  Finder({ this.skipOffstage = true });
 
-  /// Describes what the finder is looking for. The description should be
-  /// a brief English noun phrase describing the finder's pattern.
-  String get description;
+/// Provides lightweight syntax for getting frequently used semantics finders.
+///
+/// This class is instantiated once, as [CommonFinders.semantics], under [find].
+class CommonSemanticsFinders {
+  const CommonSemanticsFinders._();
 
-  /// Returns all the elements in the given list that match this
-  /// finder's pattern.
+  /// Finds an ancestor of [of] that matches [matching].
   ///
-  /// When implementing your own Finders that inherit directly from
-  /// [Finder], this is the main method to override. If your finder
+  /// If [matchRoot] is true, then the results of [of] are included in the
+  /// search and results.
+  FinderBase<SemanticsNode> ancestor({
+    required FinderBase<SemanticsNode> of,
+    required FinderBase<SemanticsNode> matching,
+    bool matchRoot = false,
+  }) {
+    return _AncestorSemanticsFinder(of, matching, matchRoot);
+  }
+
+  /// Finds a descendant of [of] that matches [matching].
+  ///
+  /// If [matchRoot] is true, then the results of [of] are included in the
+  /// search and results.
+  FinderBase<SemanticsNode> descendant({
```

Comment:
Should the return types here and elsewhere be typed as `SemanticsFinder`? Since the widget finders are also not typed FinderBase<Element>?

Reply:
I plan on eventually switching the return values of the widget finders to `FinderBase<Element>` as well, but it's a breaking change so I planned on grouping that particular aspect of this refactor with removing `Finder` completely to minimize churn over time.

Reply:
The finders in `CommonSemanticsFinders` that _could_ be referenced as `SemanticsFinder` got updated, but `ancestor` and `descendant` specifically aren't actually `SemanticsFinder`s. They're raw `FinderBase<SemanticsNode>` since their `allCandidates` implementations are based completely on the finders that are passed in.